### DISCUSSION TITLE
Can't import Application into my teleport instance

### DISCUSSION THREAD

**Discussion by chrisahx** [gravitational/teleport#9325] [Q&A] [answered]
I am trying to import teleport into my instance, but nothing shows up. I got no problem adding ssh nodes.

This is my configuration on the jenkins server i am trying to import:

teleport:
  nodename: dev-jenkins-01
  data_dir: /var/lib/teleport
  auth_token: <token>
  auth_servers:
    - cloud.example.com:3080
  ca_pin: sha256:<pin>
auth_service:
  enabled: no
ssh_service:
  enabled: yes
  labels:
    env: "dev"
proxy_service:
  enabled: no
app_service:
  apps:
  - name: "jenkins"
    uri: "http://127.0.0.1:8080"
    public_addr: "jenkins.cloud.example.com"

i keep getting this Error on the server when i am trying to run: "sudo teleport start":

ERRO [PROC:1]    App failed to establish connection to cluster: "dev-jenkins-01" [c4c4a75b-ddc2-4fe2-adea-0dcd50fd53bd] can not join the cluster with role App, the token is not valid, invalid character '<' looking for beginning of value. service/connect.go:68

On the teleport server i got the following teleport.yaml file:

teleport:
  nodename: cloud.example.local
  data_dir: /var/lib/teleport
  log:
    output: stderr
    severity: INFO
    format:
      output: text
  ca_pin: sha256:<pin>
  diag_addr: ""
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  public_addr: cloud.example.local:3025
ssh_service:
  enabled: "yes"
  labels:
    env: example
  commands:
  - name: hostname
    command: [hostname]
    period: 1m0s
proxy_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3023
  public_addr: cloud.example.com
  ssh_public_addr: cloud.example.com
  web_listen_addr: 0.0.0.0:3080
  tunnel_listen_addr: 0.0.0.0:3024
  https_keypairs: []
  acme: {}

does any one have any idea of why i can't add web applications? any help would be much appreciated.

**Comment by zen** [comment]
It would be much easier to help if you formatted the config snippets.

**Comment by webvictim** [comment, marked as answer]
You can use three backtick characters (\```) around code snippets in markdown, which will make them much easier to read.

This is what
  -  it looks like
  -  without

```
This is what
  - it looks like
  - with
```

Your actual error is likely because you haven't specified a value for `auth_token` in the config file on your agent.

You'll need to add a token with both the `node` and `app` roles (run `tctl tokens add --type=node,app` on your Teleport auth server) next to `auth_token` in your config file, then try joining the cluster again.
